File: src/WebHtv.Configuration/TvBoxConfigurationPayloadDecoder.cs
using System.Text;

namespace WebHtv.Configuration;

/// <summary>
/// Decodes the small wrapper used by several TVBox configuration endpoints.
/// Plain JSON is returned unchanged.
/// </summary>
public static class TvBoxConfigurationPayloadDecoder
{
    public static string Decode(string sourceText)
    {
        ArgumentException.ThrowIfNullOrWhiteSpace(sourceText);

        var trimmed = sourceText.Trim();
        if (!trimmed.StartsWith("jhSP", StringComparison.Ordinal))
        {
            return trimmed;
        }

        var separatorIndex = trimmed.IndexOf("**", StringComparison.Ordinal);
        if (separatorIndex < 4 || separatorIndex + 2 >= trimmed.Length)
        {
            throw new ArgumentException("The configuration wrapper is incomplete.", nameof(sourceText));
        }

        try
        {
            return Encoding.UTF8.GetString(Convert.FromBase64String(trimmed[(separatorIndex + 2)..])).Trim();
        }
        catch (FormatException exception)
        {
            throw new ArgumentException("The configuration wrapper contains invalid Base64 data.", nameof(sourceText), exception);
        }
    }
}

File: android/core/src/main/java/cn/mtplayer/core/config/TvBoxConfigurationPayloadDecoder.java
package cn.mtplayer.core.config;

import android.util.Base64;

import java.nio.charset.StandardCharsets;

/** Decodes the Base64 envelope used by some TVBox configuration endpoints. */
final class TvBoxConfigurationPayloadDecoder {
    interface Decoder {
        byte[] decode(String value);
    }

    static String decode(String source) {
        return decode(source, value -> Base64.decode(value, Base64.DEFAULT));
    }

    static String decode(String source, Decoder decoder) {
        if (source == null || source.trim().isEmpty()) {
            throw new IllegalArgumentException("配置接口返回了空内容");
        }

        String trimmed = source.trim();
        if (!trimmed.startsWith("jhSP")) {
            return trimmed;
        }

        int separator = trimmed.indexOf("**");
        if (separator < 4 || separator + 2 >= trimmed.length()) {
            throw new IllegalArgumentException("配置接口返回的加密包装不完整");
        }

        try {
            byte[] decoded = decoder.decode(trimmed.substring(separator + 2));
            if (decoded == null) {
                throw new IllegalArgumentException("配置接口返回的 Base64 数据无效");
            }
            return new String(decoded, StandardCharsets.UTF_8).trim();
        } catch (IllegalArgumentException exception) {
            throw new IllegalArgumentException("配置接口返回的 Base64 数据无效", exception);
        }
    }

    private TvBoxConfigurationPayloadDecoder() { }
}
